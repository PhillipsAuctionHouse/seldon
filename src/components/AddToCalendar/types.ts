export interface CalendarEvent {
  title: string;
  description: string;
  start: Date;
  end: Date | null;
  location: string;
  timezone: string;
}
