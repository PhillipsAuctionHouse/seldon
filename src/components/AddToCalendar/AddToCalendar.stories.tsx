import { Meta } from '@storybook/react';
import AddToCalendar, { AddToCalendarProps } from './AddToCalendar';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta = {
  title: 'Components/AddToCalendar',
  component: AddToCalendar,
} satisfies Meta<typeof AddToCalendar>;

export default meta;
export const Playground = (props: AddToCalendarProps) => <AddToCalendar {...props} />;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
Playground.args = {
  event: {
    title: 'Jewels & More: Online Auction',
    description: 'Jewels & More: Online Auction.',
    start: new Date('2025-01-27T15:13:02.59+00:00'),
    end: new Date('2025-06-18T18:15:02.59+00:00'),
    location: 'New York',
    timezone: 'America/New_York',
  },
};

Playground.argTypes = {};
