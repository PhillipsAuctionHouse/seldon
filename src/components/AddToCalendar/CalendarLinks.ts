import type { CalendarEvent } from './types';
import * as ics from 'ics';

const generateGoogleCalendarLink = (event: CalendarEvent) => {
  const start = event.start;
  const end = event.end;
  const title = encodeURIComponent(event.title);
  const description = encodeURIComponent(event.description);
  const link = `https://www.google.com/calendar/render?action=TEMPLATE&text=${title}&dates=${formatDate(start)}/${formatDate(end)}&details=${description}&location=${event.location}&ctz=${event.timezone}`;
  return link;
};

const generateOutlookOnlineLink = (event: CalendarEvent) => {
  const start = event.start;
  const end = event.end;
  const link = `https://outlook.live.com/owa/?path=/calendar/action/compose&rru=addevent&subject=${event.title}&startdt=${formatDateForOutlook(start)}&enddt=${formatDateForOutlook(end)}&body=${event.description}&location=${event.location}`;
  return link;
};

const generateYahooCalendarLink = (event: CalendarEvent) => {
  const start = event.start;
  const end = event.end;
  const link = `https://calendar.yahoo.com/?v=60&view=d&type=20&title=${event.title}&st=${formatDate(start)}&et=${formatDate(end)}&desc=${event.description}&in_loc=${event.location}`;
  return link;
};

const generateCalendarFile = (event: CalendarEvent) => {
  const { error, value } = ics.createEvent({
    title: event.title,
    description: event.description,
    start: [
      event.start.getFullYear(),
      event.start.getMonth() + 1,
      event.start.getDate(),
      event.start.getHours(),
      event.start.getMinutes(),
    ],
    end: [
      event.end.getFullYear(),
      event.end.getMonth() + 1,
      event.end.getDate(),
      event.end.getHours(),
      event.end.getMinutes(),
    ],
    location: event.location,
    startOutputType: 'local',
    startInputType: 'local',
  });

  if (error) {
    console.log(error);
  } else {
    const link = document.createElement('a');
    link.href = `data:text/calendar;charset=utf8,${value}`;
    link.download = `${event.title}.ics`;
    link.click();
  }
};

const formatDate = (date: Date) => {
  return date.toISOString().replace(/[-:.]/g, '').replace('Z', '');
};

const formatDateForOutlook = (date: Date) => {
  return date.toISOString().replace('Z', '');
};

export { generateGoogleCalendarLink, generateOutlookOnlineLink, generateYahooCalendarLink, generateCalendarFile };

// export const generateCalendarLinks = (event: CalendarEvent) => {
//   const { title, description, start, end, location, timezone } = event;

//   const formatGoogleDate = (date: Date) => {
//     return date.toISOString().replace(/[-:.]/g, '').split('.')[0];
//   };

//   const formatISODate = (date: Date) => {
//     return date.toISOString().replace(/\..+/, '');
//   };

//   const formatYahooDate = (date: Date) => {
//     return date.toISOString().replace(/[-:.]/g, '').split('.')[0];
//   };

//   const formatAppleDate = (date: Date) => {
//     return date.toISOString().replace(/[-:.]/g, '').split('.')[0] + 'Z';
//   };

//   const calendars = [
//     {
//       name: 'Google Calendar',
//       link: `https://calendar.google.com/calendar/u/0/r/eventedit?text=${encodeURIComponent(title)}&dates=${formatGoogleDate(start)}/${formatGoogleDate(end)}&ctz=${timezone}&details=${encodeURIComponent(description)}&location=${encodeURIComponent(location)}&pli=1&sf=true&output=xml`,
//     },
//     {
//       name: 'Outlook.com',
//       link: `https://outlook.office.com/calendar/0/deeplink/compose?path=/calendar/action/compose&rru=addevent&startdt=${formatISODate(start)}&enddt=${formatISODate(end)}&subject=${encodeURIComponent(title)}&location=${encodeURIComponent(location)}&body=${encodeURIComponent(description)}`,
//     },
//     {
//       name: 'Outlook',
//       link: `https://www.microsoft.com/en-us/microsoft-365/outlook/email-and-calendar-software-microsoft-outlook`,
//     },
//     {
//       name: 'Yahoo Calendar',
//       link: `https://calendar.yahoo.com/?v=60&view=d&type=20&title=${encodeURIComponent(title)}&st=${formatYahooDate(start)}&et=${formatYahooDate(end)}&desc=${encodeURIComponent(description)}&in_loc=${encodeURIComponent(location)}`,
//     },
//     {
//       name: 'Apple Calendar',
//       link: `data:text/calendar;charset=utf8,BEGIN:VCALENDAR%0D%0AVERSION:2.0%0D%0ABEGIN:VEVENT%0D%0AUID:${new Date().getTime()}%0D%0ADTSTART:${formatAppleDate(start)}%0D%0ADTEND:${formatAppleDate(end)}%0D%0ASUMMARY:${encodeURIComponent(title)}%0D%0ADESCRIPTION:${encodeURIComponent(description)}%0D%0ALOCATION:${encodeURIComponent(location)}%0D%0AEND:VEVENT%0D%0AEND:VCALENDAR`,
//     },
//     {
//       name: 'iCalendar',
//       link: `data:text/calendar;charset=utf8,BEGIN:VCALENDAR%0D%0AVERSION:2.0%0D%0ABEGIN:VEVENT%0D%0AUID:${new Date().getTime()}%0D%0ADTSTART:${formatAppleDate(start)}%0D%0ADTEND:${formatAppleDate(end)}%0D%0ASUMMARY:${encodeURIComponent(title)}%0D%0ADESCRIPTION:${encodeURIComponent(description)}%0D%0ALOCATION:${encodeURIComponent(location)}%0D%0AEND:VEVENT%0D%0AEND:VCALENDAR`,
//     },
//   ];

//   return calendars;
// };
