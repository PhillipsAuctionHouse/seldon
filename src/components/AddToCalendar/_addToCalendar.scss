@use '#scss/allPartials' as *;

.#{$px}-add-to-calendar {
  svg {
    height: 30px;
    width: 30px;
  }
  .#{$px}-icon-button {
    background: transparent;
    border: none;
    cursor: pointer;

    &:hover {
      background: transparent;
    }

    svg path {
      transition: fill 0.25s ease-in-out;
    }

    svg:hover path {
      fill: $button-hover;
    }

    &:focus-visible {
      border-radius: 5px;
      outline: 1px solid $medium-gray;
    }
  }

  &-atcb-list {
    background: #fff;
    border: 1px solid#ECEAE7;
  }

  &-atcb-item {
    padding: $spacing-xsm $spacing-sm;

    &:hover,
    &:focus-visible {
      background-color: $light-gray;
      cursor: pointer;
      outline: none;
    }
  }

  &-atcb-item-link {
    @include text($body-size2);

    background: none;
    border: none;
    cursor: pointer;
    font-variation-settings: 'wght' 400;
    outline: none;
    padding: 0;
    text-decoration: none;

    p {
      margin-bottom: 0;
    }

    &:hover {
      border: none;
      outline: none;
      text-decoration: none;
    }

    span {
      @include text($body-size2);

      font-variation-settings: 'wght' 400;
      text-decoration: none;

      &:hover,
      &:focus-visible {
        color: #000;
        outline: none;
        text-decoration: none;
      }
    }
  }
}
