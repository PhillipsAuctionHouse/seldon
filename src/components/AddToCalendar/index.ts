export { default as AddToCalendar, type AddToCalendarProps } from './AddToCalendar';
