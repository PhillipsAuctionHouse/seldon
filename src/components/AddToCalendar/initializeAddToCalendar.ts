// Because we're not allowed to export non-default exports from our components (otherwise we break HotModuleReplacement of react/refresh)
// We'll need to export any helper functions from this file
// React/refresh documentation on the issue:
// https://github.com/vitejs/vite/discussions/4583#discussioncomment-1802717
// Because we're not allowed to export non-default exports from our components (otherwise we break HotModuleReplacement of react/refresh)
// We'll need to export any helper functions from this file
// React/refresh documentation on the issue:
// https://github.com/vitejs/vite/discussions/4583#discussioncomment-1802717

declare global {
  interface Window {
    addtocalendar: {
      load: () => void;
    };

    ifaddtocalendar: number | undefined;
  }
}

export default function initializeAddToCalendar(): Promise<string> {
  return new Promise((resolve, reject) => {
    if (typeof window !== 'undefined' && window.addtocalendar) {
      if (typeof window.addtocalendar.load === 'function') {
        resolve('true');
      }
    } else {
      let intervalCount = 0;
      if (typeof window !== 'undefined' && window.ifaddtocalendar === undefined) {
        window.ifaddtocalendar = 1;
        const d = document;
        const s = d.createElement('script');
        const g = 'getElementsByTagName';
        s.type = 'text/javascript';
        s.charset = 'UTF-8';
        s.async = true;
        s.src = `${window.location.protocol === 'https:' ? 'https' : 'http'}://addtocalendar.com/atc/1.5/atc.min.js`;
        s.onload = () => {
          resolve('true');
        };
        s.onerror = () => {
          reject('Failed to load AddToCalendar script');
        };
        d[g]('head')[0].appendChild(s);
      } else {
        const checkInterval = setInterval(() => {
          intervalCount++;
          if (typeof window !== 'undefined' && window.addtocalendar) {
            clearInterval(checkInterval);
            resolve('true');
          } else if (intervalCount >= 10) {
            clearInterval(checkInterval);
            reject('AddToCalendar script failed to load');
          }
        }, 1000);
      }
    }
  });
}
