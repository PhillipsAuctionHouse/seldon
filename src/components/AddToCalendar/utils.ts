export const getTimeZone = (location: string): string => {
  const locations: { [key: string]: string } = {
    Geneva: 'Europe/Paris',
    'Hong Kong': 'Asia/Hong_Kong',
    London: 'Europe/London',
    'New York': 'America/New_York',
    Paris: 'Europe/Paris',
    Shanghai: 'Europe/London',
    Singapore: 'Asia/Singapore',
    Taipei: 'Europe/London',
  };
  return locations[location] || '';
};
