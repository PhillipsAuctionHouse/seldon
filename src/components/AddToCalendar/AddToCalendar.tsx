import { ComponentProps, forwardRef, useState, useEffect } from 'react';
import { getCommonProps, px } from '../../utils';
import classnames from 'classnames';
import { format } from 'date-fns/format';
import initializeAddToCalendar from './initializeAddToCalendar';
import { Icon } from '../Icon';
import { getTimeZone } from './utils';

// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
export interface AddToCalendarProps extends ComponentProps<'button'> {
  id?: string;
  startDateTimeOffset?: string;
  endDateTimeOffset?: string;
  timeZone?: string;
  title?: string;
  description?: string;
  location?: string;
  organizer?: string;
  organizerEmail?: string;
}
/**
 * ## Overview
 *
 * Overview of this widget
 *
 * [Figma Link](https://www.figma.com/design/OvBXAq48blO1r4qYbeBPjW/RW---Sale-Page-(PLP)?node-id=4791-30655)
 *
 * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-addtocalendar--overview)
 */
const AddToCalendar = forwardRef<HTMLButtonElement, AddToCalendarProps>(
  (
    {
      className,
      startDateTimeOffset,
      endDateTimeOffset,
      timeZone = 'UTC',
      title,
      description,
      location = 'New York',
      organizer,
      organizerEmail,
      ...props
    },
    ref,
  ) => {
    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'AddToCalendar');
    const { id } = props;
    const [status, setStatus] = useState('mounting');

    useEffect(() => {
      initializeAddToCalendar()
        .then(() => setStatus('connected'))
        .catch((error) => {
          console.error('Error initializing calendar:', error);
          setStatus('error');
        });
    }, []);

    const buttonAriaLabel = title ? `Add ${title} to calendar` : 'Add to calendar';

    const statusClassName = classnames({
      hidden: status === 'mounting' || status === 'error',
    });

    if (!startDateTimeOffset) {
      return null;
    }

    const parseDate = (str: string): Date => {
      const [year, month, date, hours, minutes, seconds] =
        str
          .match(/(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})/)
          ?.slice(1)
          .map((n) => parseInt(n, 10)) || [];
      return new Date(year, month - 1, date, hours, minutes, seconds);
    };

    const timeZoneToUse = timeZone || getTimeZone(location) || 'UTC';
    const startDateTime = parseDate(startDateTimeOffset);
    const endDateTime = endDateTimeOffset ? parseDate(endDateTimeOffset) : null;

    return (
      <button
        className={classnames(baseClassName, className)}
        {...commonProps}
        {...props}
        ref={ref}
        aria-label={buttonAriaLabel}
        disabled={status !== 'connected'}
      >
        <div id="add-to-calendar" className={statusClassName}>
          <span className="addtocalendar atc-style-icon atc-style-menu-wb">
            <a
              data-testid={`${id}-calendar-button`}
              className={`${px}-icon-button ${px}-icon-button--primary ${baseClassName}__back-button atcb-link`}
            >
              <Icon icon="CalendarAlt" />
            </a>
            <var className="atc_event">
              <var className="atc_date_start">{format(startDateTime, 'yyyy-MM-dd HH:mm:ss')}</var>
              {endDateTime && <var className="atc_date_end">{format(endDateTime, 'yyyy-MM-dd HH:mm:ss')}</var>}
              <var className="atc_timezone">{timeZoneToUse}</var>
              <var className="atc_title">{title}</var>
              <var className="atc_description">{description}</var>
              <var className="atc_location">{location}</var>
              <var className="atc_organizer">{organizer}</var>
              <var className="atc_organizer_email">{organizerEmail}</var>
            </var>
          </span>
        </div>
      </button>
    );
  },
);

AddToCalendar.displayName = 'AddToCalendar';

export default AddToCalendar;
