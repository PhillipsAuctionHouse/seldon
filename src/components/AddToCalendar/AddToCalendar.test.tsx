import { render } from '@testing-library/react';
import AddToCalendar from './AddToCalendar';

describe('AddToCalendar component', () => {
  it('renders correctly', () => {
    const props = {
      startDateTimeOffset: '2022-01-01T12:00:00',
      endDateTimeOffset: '2022-01-01T13:00:00',
      timeZone: 'UTC',
      title: 'Test Event',
      description: 'This is a test event',
      location: 'New York',
      organizer: 'John Doe',
      organizerEmail: 'johndoe@example.com',
    };

    const { container } = render(<AddToCalendar {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('renders calendar icon with correct SVG paths', () => {
    const props = {
      startDateTimeOffset: '2022-01-01T12:00:00',
      endDateTimeOffset: '2022-01-01T13:00:00',
      timeZone: 'UTC',
      title: 'Test Event',
      description: 'This is a test event',
      location: 'New York',
      organizer: 'John Doe',
      organizerEmail: 'johndoe@example.com',
    };
    const { container } = render(<AddToCalendar {...props} />);
    const svg = container.querySelector('svg');
    expect(svg).toBeInTheDocument();
    if (svg) {
      const paths = svg.querySelectorAll('path');
      expect(paths).toHaveLength(6);
    }
  });

  it('renders event details', () => {
    const props = {
      startDateTimeOffset: '2022-01-01T12:00:00',
      endDateTimeOffset: '2022-01-01T13:00:00',
      timeZone: 'UTC',
      title: 'Test Event',
      description: 'This is a test event',
      location: 'New York',
      organizer: 'John Doe',
      organizerEmail: 'johndoe@example.com',
    };
    const { container } = render(<AddToCalendar {...props} />);

    const eventDetails = container.querySelector('var.atc_event');
    expect(eventDetails).toBeInTheDocument();
    if (eventDetails) {
      const dateStart = eventDetails.querySelector('var.atc_date_start');
      expect(dateStart).toHaveTextContent('2022-01-01 12:00:00');
      const dateEnd = eventDetails.querySelector('var.atc_date_end');
      expect(dateEnd).toHaveTextContent('2022-01-01 13:00:00');
      const timezone = eventDetails.querySelector('var.atc_timezone');
      expect(timezone).toHaveTextContent('UTC');
      const title = eventDetails.querySelector('var.atc_title');
      expect(title).toHaveTextContent('Test Event');
      const description = eventDetails.querySelector('var.atc_description');
      expect(description).toHaveTextContent('This is a test event');
      const location = eventDetails.querySelector('var.atc_location');
      expect(location).toHaveTextContent('New York');
      const organizer = eventDetails.querySelector('var.atc_organizer');
      expect(organizer).toHaveTextContent('John Doe');
      const organizerEmail = eventDetails.querySelector('var.atc_organizer_email');
      expect(organizerEmail).toHaveTextContent('johndoe@example.com');
    }
  });

  it('renders Add to Calendar button with correct class names', () => {
    const props = {
      startDateTimeOffset: '2022-01-01T12:00:00',
      endDateTimeOffset: '2022-01-01T13:00:00',
      timeZone: 'UTC',
      title: 'Test Event',
      description: 'This is a test event',
      location: 'New York',
      organizer: 'John Doe',
      organizerEmail: 'johndoe@example.com',
    };
    const { container } = render(<AddToCalendar {...props} />);
    const button = container.querySelector('span.addtocalendar');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('atc-style-icon');
    expect(button).toHaveClass('atc-style-menu-wb');
  });
});
