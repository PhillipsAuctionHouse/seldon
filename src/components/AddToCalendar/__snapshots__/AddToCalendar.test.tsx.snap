// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddToCalendar component > renders correctly 1`] = `
<div>
  <button
    aria-label="Add Jewels & More: Online Auction to calendar"
    class="seldon-add-to-calendar"
    data-testid="add-to-calendar"
    disabled=""
  >
    <div
      class="hidden"
      id="add-to-calendar"
    >
      <span
        class="addtocalendar atc-style-icon atc-style-menu-wb"
      >
        <a
          class="seldon-icon-button seldon-icon-button--primary seldon-add-to-calendar__back-button atcb-link"
          data-testid="undefined-calendar-button"
        >
          <div
            class="seldon-icon seldon-icon-calendar-alt"
          >
            <svg
              aria-labelledby="svg-calendar-alt"
              color="$pure-black"
              data-testid="icon-calendar-alt"
              fill="none"
              role="img"
              viewBox="0 0 28 28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-calendar-alt"
              >
                SvgCalendarAlt
              </title>
              <path
                d="M7 12.25A.875.875 0 1 0 7 14h1.75a.875.875 0 0 0 0-1.75zM13.125 12.25a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75zM19.25 12.25a.875.875 0 1 0 0 1.75H21a.875.875 0 0 0 0-1.75zM7 17.5a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75zM13.125 17.5a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75z"
                fill="$pure-black"
              />
              <path
                clip-rule="evenodd"
                d="M1.75 7a3.5 3.5 0 0 1 3.5-3.5h17.5a3.5 3.5 0 0 1 3.5 3.5v14a3.5 3.5 0 0 1-3.5 3.5H5.25a3.5 3.5 0 0 1-3.5-3.5zM24.5 9.625V21a1.75 1.75 0 0 1-1.75 1.75H5.25A1.75 1.75 0 0 1 3.5 21V9.625zm0-1.75V7a1.75 1.75 0 0 0-1.75-1.75H5.25A1.75 1.75 0 0 0 3.5 7v.875z"
                fill="$pure-black"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </a>
        <var
          class="atc_event"
        >
          <var
            class="atc_date_start"
          >
            2025-01-27 15:13:02
          </var>
          <var
            class="atc_date_end"
          >
            2025-06-27 15:13:02
          </var>
          <var
            class="atc_timezone"
          >
            America/New_York
          </var>
          <var
            class="atc_title"
          >
            Jewels & More: Online Auction
          </var>
          <var
            class="atc_description"
          >
            Jewels & More: Online Auction.
          </var>
          <var
            class="atc_location"
          >
            New York
          </var>
          <var
            class="atc_organizer"
          >
            Conference Organizers
          </var>
          <var
            class="atc_organizer_email"
          >
            test@example.com
          </var>
        </var>
      </span>
    </div>
  </button>
</div>
`;

exports[`AddToCalendar component > renders correctly with startDateTimeOffset 1`] = `
<div>
  <button
    aria-label="Add Test Event to calendar"
    class="seldon-add-to-calendar"
    data-testid="add-to-calendar"
    disabled=""
  >
    <div
      class="hidden"
      id="add-to-calendar"
    >
      <span
        class="addtocalendar atc-style-icon atc-style-menu-wb"
      >
        <a
          class="seldon-icon-button seldon-icon-button--primary seldon-add-to-calendar__back-button atcb-link"
          data-testid="undefined-calendar-button"
        >
          <div
            class="seldon-icon seldon-icon-calendar-alt"
          >
            <svg
              aria-labelledby="svg-calendar-alt"
              color="$pure-black"
              data-testid="icon-calendar-alt"
              fill="none"
              role="img"
              viewBox="0 0 28 28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-calendar-alt"
              >
                SvgCalendarAlt
              </title>
              <path
                d="M7 12.25A.875.875 0 1 0 7 14h1.75a.875.875 0 0 0 0-1.75zM13.125 12.25a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75zM19.25 12.25a.875.875 0 1 0 0 1.75H21a.875.875 0 0 0 0-1.75zM7 17.5a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75zM13.125 17.5a.875.875 0 1 0 0 1.75h1.75a.875.875 0 0 0 0-1.75z"
                fill="$pure-black"
              />
              <path
                clip-rule="evenodd"
                d="M1.75 7a3.5 3.5 0 0 1 3.5-3.5h17.5a3.5 3.5 0 0 1 3.5 3.5v14a3.5 3.5 0 0 1-3.5 3.5H5.25a3.5 3.5 0 0 1-3.5-3.5zM24.5 9.625V21a1.75 1.75 0 0 1-1.75 1.75H5.25A1.75 1.75 0 0 1 3.5 21V9.625zm0-1.75V7a1.75 1.75 0 0 0-1.75-1.75H5.25A1.75 1.75 0 0 0 3.5 7v.875z"
                fill="$pure-black"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </a>
        <var
          class="atc_event"
        >
          <var
            class="atc_date_start"
          >
            2022-01-01 12:00:00
          </var>
          <var
            class="atc_date_end"
          >
            2022-01-01 13:00:00
          </var>
          <var
            class="atc_timezone"
          >
            UTC
          </var>
          <var
            class="atc_title"
          >
            Test Event
          </var>
          <var
            class="atc_description"
          >
            This is a test event
          </var>
          <var
            class="atc_location"
          >
            New York
          </var>
          <var
            class="atc_organizer"
          >
            John Doe
          </var>
          <var
            class="atc_organizer_email"
          >
            johndoe@example.com
          </var>
        </var>
      </span>
    </div>
  </button>
</div>
`;
